### PR TITLE
Remove old profiles, use JDK 17, upgrade to latest plugin and platform

### PR DESCRIPTION
**Pull request by londonbrown**
*Issue #, if available:* #46

*Description of changes:*
- Remove older profiles to reduce clutter db43e0e748f574beab34b4569fb1e9dd2fe5f0f0
- Update gradle-intellij-plugin to 1.12.0 (latest) 60aef7587b49b23037f71afb542e8539b249f3c9
- Use JDK 17 during build as required by build 223+ [[blog]](https://blog.jetbrains.com/platform/2022/08/intellij-project-migrates-to-java-17/?_gl=1*16a5aek*_ga*MjI1NDA1NjYwLjE2MzY1MDQzNDE.*_ga_9J976DJZ68*MTY3NDYzNjgwMy42OTEuMS4xNjc0NjM5MzQyLjAuMC4w&_ga=2.109930242.642355864.1674636804-225405660.1636504341) (doesn't effect output jar, it's still jdk 11) 70c5abdccef496683c71048860caf39104100379
- Support 2023.1 EAP a695cf58a479e9cc0cfc8c24a7e83d60d350cd4b

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by londonbrown** [comment]
Apologies for the previous failed builds. Needed to use PR to test github actions were working correctly and ran into some issues with upgrading the dependencies at first. But builds are now succeeding as expected for current stable version and the new EAP version this PR is about.

**Comment by londonbrown** [comment]
Apologies for the previous failed builds. Needed to use PR to test github actions were working correctly and ran into some issues with upgrading the dependencies at first. But builds are now succeeding as expected for current stable version and the new EAP version this PR is about.

File: src/IC-231/kotlin/com/amazon/ion/plugin/intellij/formatting/blocks/IonSExpressionBlock.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.amazon.ion.plugin.intellij.psi.IonTypes
+import com.amazon.ion.plugin.intellij.psi.isOneLiner
+import com.amazon.ion.plugin.intellij.utils.elementIsA
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.diagnostic.debug
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.psi.tree.IElementType
+import org.jetbrains.kotlin.idea.base.psi.getLineNumber
+
+// import org.jetbrains.kotlin.idea.core.util.getLineNumber
```
Comment (popematt) [inline src/IC-231/kotlin/com/amazon/ion/plugin/intellij/formatting/blocks/IonSExpressionBlock.kt:14]:
```suggestion
```

File: src/IC-231/kotlin/com/amazon/ion/plugin/intellij/psi/PsiElementExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.amazon.ion.plugin.intellij.psi
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.base.psi.getLineCount
+
+// import org.jetbrains.kotlin.idea.core.util.getLineCount
```
Comment (popematt) [inline src/IC-231/kotlin/com/amazon/ion/plugin/intellij/psi/PsiElementExtensions.kt:6]:
```suggestion
```

**Review by popematt — APPROVED** [review]
Looks good, thanks! There's a few stray comments to clean up, then we're good to go.

If you feel like updating to `setup-java@v3` you could, but it's also fine if you just leave it the way it is.

**Pull request merged**